File: Number of Students Unable to Eat Lunch.cpp
class Solution 
{
public:
    int countStudents(vector<int>& students, vector<int>& sandwiches) 
    {
        queue<int>stud;
        stack<int>sand;
        reverse(sandwiches.begin(), sandwiches.end());
        for(int i = 0; i < students.size(); i++)
        {
            stud.push(students[i]);
            sand.push(sandwiches[i]);
        }
        int count = 0;
        while(!stud.empty())
        {
            if(count == stud.size()) return stud.size();
            if(stud.front() == sand.top())
            {
                stud.pop();
                sand.pop();
                count = 0;
            }
            else
            {
                stud.push(stud.front());
                stud.pop();
                count++;
            }
        }
        return 0;
    }
};

File: Number of Students Unable to Eat Lunch.py
class Solution(object):
    def countStudents(self, students, sandwiches):
        stud = deque(students)
        sand = deque(reversed(sandwiches))
        count = 0
        while stud:
            if count == len(stud):
                return len(stud)
            if stud[0] == sand[-1]:
                stud.popleft()
                sand.pop()
                count = 0
            else:
                stud.append(stud.popleft())
                count += 1
                
        return 0